### PR TITLE
feat: add rainbowkit integration

### PR DESCRIPTION
**Pull request by bigint**
### <samp>🤖 Generated by Copilot at f29d7c6</samp>

This pull request refactors the `web` app to use a custom `Web3Provider` component instead of the `WagmiConfig` component, and adds the `@rainbow-me/rainbowkit` library for UI components and hooks. It also updates the `pnpm-lock.yaml` file and removes some unused imports from the `ui` package.

## Related issues

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at f29d7c6</samp>

*  Add `@rainbow-me/rainbowkit` dependency to `web` app and import its stylesheets in `_app.tsx` ([link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R47), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-a3569d521976a5ee9e97e2b2e005bf7423bc96bb49397b3adfb09cde5862bf93R2-R3), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR156-R158), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5758-R5780))
*  Replace `WagmiConfig` component with custom `Web3Provider` component in `Providers.tsx` and create new file `Web3Provider.tsx` to configure web3 context and connectors ([link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-f7e122c26f9b485a5a519e88e671db9096185dd146ff170dcd0a37c71f57b573L62-R31), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-f7e122c26f9b485a5a519e88e671db9096185dd146ff170dcd0a37c71f57b573L73-R42), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-a3dd828a17894321fdb82d5140b9e1061b82bae49196a4f6c01447fb4c586431R1-R43))
*  Remove unused imports and variables from `Providers.tsx` related to `wagmi` library ([link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-f7e122c26f9b485a5a519e88e671db9096185dd146ff170dcd0a37c71f57b573L1-R1), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-f7e122c26f9b485a5a519e88e671db9096185dd146ff170dcd0a37c71f57b573L17-L22), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-f7e122c26f9b485a5a519e88e671db9096185dd146ff170dcd0a37c71f57b573L28-R17))
*  Remove unused imports from `HelpTooltip.tsx` and `Tooltip.tsx` related to `tippy.js` library ([link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-2c23bd5143e2012ffbfffb5a2b746cac1e633bc434f848e5701af124ff7f045fL1-L2), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-0524c51f0c05cf856c330f121f2e77a3d692c6b22c42572b9c63d465d8d06827L1-L2))
*  Add and update transitive dependencies of `@rainbow-me/rainbowkit` and `@vanilla-extract/css` libraries in `pnpm-lock.yaml` ([link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2300-R2303), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR6922-R6955), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR7786-R7789), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8598-R8602), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8789-R8793), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8725), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8911-R8919), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR11570-R11575), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR12295-R12298), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR12873-R12883), [link](https://github.com/lensterxyz/lenster/pull/3423/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR13108-R13126))

## Emoji

🌈🌐🧹

### CODE REVIEW

File: apps/web/next.config.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ const headers = [{ key: 'Cache-Control', value: 'public, max-age=3600' }];
 const nextConfig = {
   transpilePackages: ['data'],
   reactStrictMode: false,
+  webpack: (config) => {
```
Comment (sasicodes) [inline apps/web/next.config.js:7]: why need this? 🤔

**Review by sasicodes — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Shared/Auth/index.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      if (!signature) {
+        return toast.error(t`Invalid Signature!`);
+      }
+      const result = await authenticate({
+        variables: { request: { address, signature } }
+      });
+      const accessToken = result.data?.authenticate.accessToken;
+      const refreshToken = result.data?.authenticate.refreshToken;
+      signIn({ accessToken, refreshToken });
+      const { data: channelsData } = await getProfiles({
```
Comment (sasicodes) [inline apps/web/src/components/Shared/Auth/index.tsx:101]:
```suggestion
      const { data: profilesData } = await getProfiles({
```

File: apps/web/src/components/Shared/Auth/index.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        }
+      });
+      if (
+        !channelsData?.profiles ||
+        channelsData?.profiles?.items.length === 0
+      ) {
+        setCurrentProfile(null);
+        setProfileId(null);
+      } else {
+        const channels = channelsData?.profiles?.items as Profile[];
```
Comment (sasicodes) [inline apps/web/src/components/Shared/Auth/index.tsx:113]:
```suggestion
        const profiles = channelsData?.profiles?.items as Profile[];
```

File: apps/web/src/components/Shared/Auth/index.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        const defaultChannel = channels.find((channel) => channel.isDefault);
```
Comment (sasicodes) [inline apps/web/src/components/Shared/Auth/index.tsx:114]:
```suggestion
        const defaultProfile = profiles.find((profile) => profile.isDefault);
```

File: apps/web/src/components/Shared/Auth/index.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        setProfiles(channels);
```
Comment (sasicodes) [inline apps/web/src/components/Shared/Auth/index.tsx:115]:
```suggestion
        setProfiles(profiles);
```

File: apps/web/src/components/Shared/Auth/index.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        setCurrentProfile(defaultChannel ?? channels[0]);
+        setProfileId(defaultChannel?.id ?? channels[0].id);
```
Comment (sasicodes) [inline apps/web/src/components/Shared/Auth/index.tsx:117]:
```suggestion
        setCurrentProfile(defaultProfile ?? profiles[0]);
        setProfileId(defaultProfile?.id ?? profiles[0].id);
```

**Review by sasicodes — CHANGES_REQUESTED** [review]

File: apps/web/next.config.js (hunk 1 above)
Comment (bigint) [reply apps/web/next.config.js:7]: Node 18 😢

Comment (sasicodes) [reply apps/web/next.config.js:7]: myself using v18 in lenstube

**Comment by bigint** [comment]
We don't want now 🙇🏼

**Pull request closed without merging**